### PR TITLE
Trigger traffic signals [OTP-1125]

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds configurations for triggering actions when someone approaches a particular segment defined by coordinates.
The trigger included in this PR is for the API that controls select traffic signals in the Atlanta region. Note that such requests can take a few seconds to complete, so I added a mechanism to discard a call to the signal API if another one has not returned yet.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    public String getUrl(String signalId, String crossingId, boolean extended) {
+        return host + String.format(path, signalId, crossingId) + (extended ? "?extended=true" : "");
+    }
+
+    public Map<String, String> getHeaders() {
+        return Map.of("X-API-KEY", key);
+    }
+
+    public String getBody() {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java]: So no body is required? Seems like this method could be removed?

File: src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                } else {
+                    String pathAndQuery = getUrl(signalId, crossingId, extended);
+                    var httpResponse = HttpUtils.httpRequestRawResponse(
+                        URI.create(pathAndQuery),
+                        30,
+                        HttpMethod.POST,
+                        getHeaders(),
+                        getBody()
+                    );
+                    if (httpResponse.status == 200) {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java]: Could replace with HttpStatus.OK_200.

**Review by br648 — APPROVED** [review]
Couple of minors, but looks good to me. We could probably combined the trip actions and bus notifier actions in a later PR.

File: src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java]: Updated in https://github.com/ibi-group/otp-middleware/pull/235/commits/176b5a9acdf5fe2ce09546b71e3673b77470860a.

File: src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/triptracker/interactions/UsGdotGwinnettTrafficSignalNotifier.java]: Yeah I'll remove for now, see https://github.com/ibi-group/otp-middleware/pull/235/commits/176b5a9acdf5fe2ce09546b71e3673b77470860a

**Review by amy-corson-ibigroup — APPROVED** [review]
Sorry for how long this took me but I think it looks good!

**Pull request merged**